packages/block-library/src/comments-query-loop/edit.js

```diff
@@ -3,47 +3,61 @@
  */
 import {
 	BlockControls,
-	InspectorControls,
 	useBlockProps,
 	useInnerBlocksProps,
 } from '@wordpress/block-editor';
-import { SelectControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import QueryToolbar from './toolbar';
+import CommentsToolbar from './edit/toolbar';
+import CommentsInspectorControls from './edit/comments-inspector-controls';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
 
 const TEMPLATE = [ [ 'core/comment-template' ] ];
 
 export default function CommentsQueryLoopEdit( { attributes, setAttributes } ) {
-	const { perPage, tagName: TagName } = attributes;
+	const { tagName: TagName, order } = attributes;
 
 	const blockProps = useBlockProps();
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: TEMPLATE,
 	} );
 
+	const settingsOrder = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+
+		const { comment_order: commentOrder } =
+			getEntityRecord( 'root', 'site' ) || {};
+
+		// We cannot get the order of the discussion settings page if we are not an admin user.
+		// Without the order defined, the default on the API is DESC
+		// with means show the newest comments first.
+		// By default, in discussion settings, the order is ASC.
+		// So the frontend is showing older comments first, while the editor is showing newer ones.
+		if ( commentOrder ) return commentOrder;
+	} );
+
+	// If we are an admin user and have access to discussion settings,
+	// we set them as the default one if the order has not been defined before.
+
+	if ( settingsOrder && order === null ) {
+		setAttributes( { order: settingsOrder } );
```

Comment:
We need to wrap this in a `useEffect` because it causes side effects.

Reply:
Actually, I think that we should remove this code because we don't want to save the current default order as an attribute for the block. We should only use the value in the UI when the user wants to change the default value – a sort of fallback or starting point depending on how you think about it.

Reply:
The reason for having this is to show a default position for the toggle, depending on what is defined on the settings version. Right now is not working if you are not an admin.

Maybe we could add an option of "Inherit from settings", default enabled. And, in case you disable it, we can set a default order attribute.

Meanwhile, I updated the code in order not to cause side effects or mess with undo buttons.